### PR TITLE
[SPARK-35021][SQL] Group exception messages in connector/catalog

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR group exception messages in `sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog`.

### Why are the changes needed?
It will largely help with standardization of error messages and its maintenance.

### Does this PR introduce _any_ user-facing change?
No. Error messages remain unchanged.

### How was this patch tested?
No new tests - pass all original tests to make sure it doesn't break any existing behavior.

### CODE REVIEW

**Comment by beliefer** [comment]
retest this please

**Comment by beliefer** [comment]
ping @allisonwang-db

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 1]
```diff
@@ -1351,4 +1352,39 @@ private[spark] object QueryCompilationErrors {
...
+
+  def cannotPartitionByNestedColumnError(reference: NamedReference): Throwable = {
+    new AnalysisException(s"Cannot partition by nested column: $reference")
+  }
+
+  def cannotUseCatalogError(plugin: CatalogPlugin, msg: String): Throwable = {
+    new AnalysisException(s"Cannot use catalog ${plugin.name}: $msg")
+  }
+
+  def invalidIdentifierAsItHasMoreThanTwoNamePartsError(
```
Comment (allisonwang-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:1374]: How about `identifierHavingMoreThanTwoNamePartsError`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 2]
```diff
@@ -1351,4 +1352,39 @@ private[spark] object QueryCompilationErrors {
...
+
+  def emptyMultipartIdentifierError(): Throwable = {
+    new AnalysisException("multi-part identifier cannot be empty.")
+  }
+
+  def cannotCreateTablesWithNullTypeError(): Throwable = {
+    new AnalysisException(s"Cannot create tables with ${NullType.simpleString} type.")
+  }
+
+  def functionUnsupportedInV1CatalogError(): Throwable = {
```
Comment (allisonwang-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:1387]: functionUnsupportedInV2CatalogError

**Comment by beliefer** [comment]
ping @allisonwang-db

**Review by allisonwang-db — APPROVED** [review]

**Comment by beliefer** [comment]
cc @cloud-fan

**Comment by cloud-fan** [comment]
@beliefer can you fix the conflicts? thanks!

**Comment by beliefer** [comment]
> @beliefer can you fix the conflicts? thanks!

OK

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**